lib/network/kmsAWS/Client.ts

```diff
@@ -199,31 +194,24 @@ export default class Client implements KMSInterface {
         const masterKeyId = getKeyIdFromArn(masterKeyIdOrArn);
         logger.debug("AWS KMS: generating data key", { cryptoScheme, masterKeyId, masterKeyIdOrArn });
         assert.strictEqual(cryptoScheme, 1);
-
         const params = {
             KeyId: masterKeyId,
-            KeySpec: 'AES_256',
+            KeySpec: 'AES_256' as const,
         };
-
-        this.client.generateDataKey(params, (err: AWSError, data) => {
-            if (err) {
-                const error = arsenalErrorAWSKMS(err);
-                logger.error('AWS KMS: failed to generate data key', { err });
-                cb(error);
-                return;
-            }
-
+        this.client.send(new GenerateDataKeyCommand(params)).then(data => {
             if (!data) {
                 const error = arsenalErrorAWSKMS("failed to generate data key: empty response");
                 logger.error("AWS KMS: failed to generate data key: empty response");
                 cb(error);
                 return;
             }
-
             const isolatedPlaintext = this.safePlaintext(data.Plaintext as Buffer);
-
             logger.debug('AWS KMS: data key generated');
             cb(null, isolatedPlaintext, Buffer.from(data.CiphertextBlob as Uint8Array));
+        }).catch(err => {
```

Comment:
is typescript able to "understand" the type of error automatically?
if not, we should still keep the type in prototype...

```suggestion
        }).catch(err: AWSError => {
```
or (if there is no detail on type, in AWS)
```suggestion
        }).catch(err: error => {
```

(just posting once, applies in multiple places...)